cpp/libtab.cpp

```diff
@@ -4,33 +4,155 @@
 
 #include "libtab.h"
 #include "crouzeix-raviart.h"
-#include "finite-element.h"
 #include "lagrange.h"
 #include "nedelec.h"
-#include "polynomial-set.h"
+#include "polyset.h"
 #include "raviart-thomas.h"
 #include "regge.h"
-#include <map>
+#include <iostream>
+
+#define str_macro(X) #X
+#define str(X) str_macro(X)
 
 using namespace libtab;
 
 //-----------------------------------------------------------------------------
 libtab::FiniteElement libtab::create_element(std::string family,
                                              std::string cell, int degree)
 {
-  const std::map<std::string, std::function<FiniteElement(cell::Type, int)>>
-      create_map = {{cr::family_name, &cr::create},
-                    {dlagrange::family_name, &dlagrange::create},
-                    {lagrange::family_name, &lagrange::create},
-                    {nedelec::family_name, &nedelec::create},
-                    {nedelec2::family_name, &nedelec2::create},
-                    {rt::family_name, &rt::create},
-                    {regge::family_name, &regge::create}};
-
-  auto create_it = create_map.find(family);
-  if (create_it == create_map.end())
+  if (family == "Lagrange")
```

Comment:
I find the const map neater than if/elseif

Reply:
The mucking about with maps to function pointers is a bit ugly.

Reply:
`else if` is ugly. matter of opinion.